### PR TITLE
buildroute use lnd default

### PR DESCRIPTION
**Pull request by ziggie1984**
Fixes https://github.com/lightningnetwork/lnd/issues/8380.

Now the lncli does not provide a default value but we check on the lnd side and if its not provided we use the lnd default.

Moreover improves the docs.

### CODE REVIEW

File: cmd/lncli/cmd_payments.go [hunk 1]
```diff
@@ -1556,7 +1555,15 @@ var buildRouteCommand = cli.Command{
 	Name:     "buildroute",
 	Category: "Payments",
 	Usage:    "Build a route from a list of hop pubkeys.",
-	Action:   actionDecorator(buildRoute),
+	Description: `
+	Builds a sphinx route for the supplied hops (public keys). Make sure to
+	use a custom final_cltv_delta to create the route depending on the
+	restrictions in the invoice otherwise LND will use its default specified
+	via the bitcoin.timelockdelta setting (default 80).
```
Comment (guggero) [inline cmd/lncli/cmd_payments.go:1562]: Should we specifically mention the `INCORRECT_OR_UNKNOWN_PAYMENT_DETAILS` error here perhaps?

File: lnrpc/routerrpc/router_server.go [hunk 2]
```diff
@@ -999,6 +999,11 @@ func (s *Server) BuildRoute(ctx context.Context,
 		payAddr = &backingPayAddr
 	}
 
+	if req.FinalCltvDelta == 0 {
+		//nolint:lll
+		req.FinalCltvDelta = int32(s.cfg.RouterBackend.DefaultFinalCltvDelta)
```
Comment (guggero) [inline lnrpc/routerrpc/router_server.go:1004]: nit: could use this formatting instead:
```
		req.FinalCltvDelta = int32(
			s.cfg.RouterBackend.DefaultFinalCltvDelta,
		)
```

File: lnrpc/routerrpc/router_server.go [hunk 3]
```diff
@@ -999,6 +999,11 @@ func (s *Server) BuildRoute(ctx context.Context,
 		payAddr = &backingPayAddr
 	}
 
+	if req.FinalCltvDelta == 0 {
```
Comment (guggero) [inline lnrpc/routerrpc/router_server.go:1002]: Nice to unify this across the CLI and RPC :+1:

**Review by guggero — COMMENTED** [review]
Great improvement, thanks for the PR!

**Review by guggero — APPROVED** [review]
Nice, LGTM :tada:

**Review by yyforyongyu — APPROVED** [review]
Thanks for the PR ⛵

**Pull request merged**